### PR TITLE
toybox: unbreak on darwin

### PR DESCRIPTION
**Pull request by Atemu**
Actually didn't need an SDK update but was just missing iconv which Darwin's libc doesn't include apparently.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Atemu** [comment]
Whoops, totally forgot x86_64-darwin uses a different SDK by default. However, calling it with `darwin.apple_sdk_11_0.callPackage` somehow does not make a difference: It still tries to use `/nix/store/dm35436lwr5dyk6izqpjrhv6w5gx9gxx-Libsystem-1238.60.2` which is the same as regular `callPackage`?  
That shouldn't happen. It should be Libsystem 1292.50.1 on the newer SDK.

Opened a bug report above.

File: pkgs/tools/misc/toybox/default.nix [hunk 1]
```diff
@@ -20,6 +20,9 @@ stdenv.mkDerivation rec {
   depsBuildBuild = [ buildPackages.stdenv.cc ]; # needed for cross
```
Comment (wegank) [inline pkgs/tools/misc/toybox/default.nix:20]: This is what causes the behaviour in #210923, actually. Should be good when using `stdenv.cc` on x86_64-darwin.

Comment (Atemu) [reply pkgs/tools/misc/toybox/default.nix:20]: Oh man, nice catch!

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -12418,7 +12418,9 @@ with pkgs;
 
   toxvpn = callPackage ../tools/networking/toxvpn { };
 
-  toybox = callPackage ../tools/misc/toybox { };
+  toybox = darwin.apple_sdk_11_0.callPackage ../tools/misc/toybox {
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix:12421]: Is this available on Linux as well?

Comment (winterqt) [reply pkgs/top-level/all-packages.nix:12421]: It's a no-op on Linux.

Comment (wegank) [reply pkgs/top-level/all-packages.nix:12421]: https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/platform-notes.chapter.md

**Comment by Atemu** [comment]
Sorry for the confusion but I forgot to push the fix adressing @wegank's comment yesterday. Not a big problem, I'll just make a follow-up PR.

**Comment by Mic92** [comment]
> Sorry for the confusion but I forgot to push the fix adressing @wegank's comment yesterday. Not a big problem, I'll just make a follow-up PR.

I missed this one as well.

**Pull request merged**